mne/viz/_figure.py

```diff
@@ -1773,6 +1773,8 @@ def _load_data(self, start=None, stop=None):
         if 'raw' in (self.mne.instance_type, self.mne.ica_type):
             return self.mne.inst[:, start:stop]
         else:
+            if not self.mne.inst.preload:
+                self.mne.inst.load_data()
```

Comment:
Better than this would probably be `get_data` geting just the part you need. It can drastically reduce memory requirements. I think it's worth going through the indexing gymnastics to figure it out.

In particular you probably need to do something like `mne.inst.drop_bad()` then look at `len(inst)` and `len(epochs.times)` and figure out which epochs you need, `get_data` for them, then concat, then subselect (trim beginning/end where necessary).

Then in your smoke test below you can `assert epochs._data is None` before and after the `plot` call

Reply:
@larsoner I've done as you suggest. Only downside is that now the console prints 

```
Loading data for 5 events and 421 original time points ...
Loading data for 6 events and 421 original time points ...
Loading data for 7 events and 421 original time points ...
Loading data for 8 events and 421 original time points ...
```
every time you scroll or change the number of visible channels. I can live with it, but might be worth considering whether `epochs.get_data` should get a `verbose` kwarg (in a separate PR).

Reply:
You can also just do `with mne.use_log_level('error'):` or `with mne.use_log_level('warn')` here if you want. It does the same thing as `verbose` actually. And even adding `verbose` is a pretty trivial change so you can make it here.

But I'm fine with another PR, too -- I'll merge if you don't push a commit before Circle comes back green :)

Reply:
... looks like that time has already passed

Reply:
... and actually it looks like I should at least wait for Azure to come back happy so that the `viz` tests run